src/macro.jl

```diff
@@ -119,7 +119,7 @@ _FASTMATH = Ref(true)
 _THREADS = Ref{Any}(true)
 _GRAD = Ref{Any}(:Base)
 _AVX = Ref{Any}(true)
-_CUDA = Ref{Any}(256)
+_CUDA = Ref{Any}(nothing)
```

Comment:
Maybe this should be `true` to mean auto (like `threads=true` now) and higher numbers to mean specific settings? Partly to avoid re-writing the function `checklegal` just below.
```suggestion
_CUDA = Ref{Any}(true)
```